components/pages/searchPageLayout/SearchPageLayout.tsx

```diff
@@ -99,24 +126,73 @@ const SearchPageLayout = ({ searchQuery }: { searchQuery?: string }) => {
       },
     },
     {
-      enabled: q?.length > 0,
+      refetchOnWindowFocus: false,
+      enabled: currentQueryString?.length > 0,
     }
   )
 
   const facetData = dataFacets?.search?.facets
-  const searchData = data?.search
+
+  const handleLoadMore = () => {
+    const totalPages = Math.ceil((data?.pages?.[0]?.search.hitcount ?? 0) / SEARCH_RESULTS_LIMIT)
+
+    if (currentPage < totalPages) {
+      fetchNextPage()
+      setCurrentPage(currentPage + 1)
+    }
+  }
+
+  useEffect(() => {
+    if (isInView) {
+      handleLoadMore()
+    }
+  }, [isInView])
+
+  // TODO: consider finding a better way to control fetching of data without using the useEffects below
+  useEffect(() => {
+    const page = data?.pages.length || 0
+    setCurrentPage(page)
+  }, [data?.pages])
+
+  useEffect(() => {
+    setCurrentPage(0)
+    setCurrentQueryString(q)
+  }, [q])
+
+  useEffect(() => {
+    const isFilterMatching = JSON.stringify(facetFilters) === JSON.stringify(facetsForSearchRequest)
+    if (!isFilterMatching) {
+      setFacetFilters(facetsForSearchRequest)
+      setCurrentPage(0)
+    }
+  }, [facetsForSearchRequest])
 
   return (
     <div className="content-container">
-      <h1 className="mt-[88px] text-typo-heading-2">
-        {`Viser resultater for "${q}" ${data?.search.hitcount ? "(" + data?.search.hitcount + ")" : ""}`}
-      </h1>
+      <h1 className="mt-[88px] text-typo-heading-2">{`Viser resultater for "${q}" ${data?.pages?.[0]?.search.hitcount ? "(" + data?.pages?.[0]?.search.hitcount + ")" : ""}`}</h1>
+      {/* TODO: add ghost loading and cleanup the code below  */}
       {isLoadingFacets && <p>isLoadingFacets...</p>}
       {!facetData?.length && <p>Ingen filter</p>}
       {facetData && facetData?.length > 0 && <SearchFilterBar facets={dataFacets.search.facets} />}
       {isLoading && <p>isLoading...</p>}
-      {searchData?.hitcount === 0 && <p>Ingen søgeresultat</p>}
-      {searchData?.works && <SearchResults works={searchData.works} />}
+
+      {data?.pages?.[0]?.search.hitcount === 0 && <p>Ingen søgeresultat</p>}
```

Comment:
This is mostly a matter of preference but I am not of repeating the same optional chaining. If we can resolve before the rendering if there are any `data.pages` it is more clean to me. Instead of asking: "do we have data.pages now and do we have dapa.pages now and..." :)